MaskedSelect.lua

```diff
@@ -0,0 +1,56 @@
+local MaskedSelect, parent = torch.class('nn.MaskedSelect', 'nn.Module')
+
+--[[ Sets the provided mask value for the module. ]]
+function MaskedSelect:__init(mask)
+  -- mask must be a tensor, convert to byte tensor
+  parent.__init(self)
+  self.mask = mask
+  self.maskIndices =
+    torch.range(1, mask:size():prod()):double():resize(mask):maskedSelect(mask)
+end
+
+--[[ Performs maskedSelect operation. ]]
+function MaskedSelect:updateOutput(input)
+  self.output = input:maskedSelect(self.mask)
+  return self.output
+end
+
+--[[ Reverse maps unmasked gradOutput back to gradInput. ]]
+function MaskedSelect:updateGradInput(input, gradOutput)
+  self.gradInput = torch.Tensor(input:size():prod())
+  local i = 1
+  local f = function(x)
+    self.gradInput[x] = self.gradOutput[i]
+    i = i + 1
+  end
+  self.maskIndices:apply(f)
```

Comment:
is this part equivalent to [`gather`](https://github.com/torch/torch7/blob/master/doc/tensor.md#tensor-gatherdim-index) ? 
The problem with the `:apply` function is that it doesn't work on GPU.